### PR TITLE
feat(stream-tile): use dids capability iss as controller when capabil…

### PR DESCRIPTION
**Pull request by zachferland**
~~DRAFT - waits on DIDs package~~

Reference [SIWE+ spec](https://www.notion.so/threebox/CACAO-DID-PKH-SIWE-Client-Libraries-5ce1f1f6e8c7417683c2b63e07f91cef#741885cca9d94532afee11a46621ab7d) for motivation. Allows devs to use DIDs w/ capability and ceramic without having to know an implementation detail of CACAO (ie controller is cap issuer, and did w/cap is the signer)

### CODE REVIEW

**Review by stbrody — COMMENTED** [review]
makes sense to me, but I'll defer to @ukstv for final approval

**Review by ukstv — CHANGES_REQUESTED** [review]
In the current iteration, the PR _replaces_ existing tests while it _adds_ new functionality.

IMO, this PR should add few new tests rather than changing old ones.

Otherwise, LGTM. The core change improves DX a lot.

**Comment by zachferland** [comment]
yeah considered, but agree, restored existing tests, added new now

File: packages/canary-integration/src/__tests__/capability.test.ts [hunk 1]
```diff
@@ -315,4 +318,56 @@ describe('CACAO Integration test', () => {
...
+      const deterministicDocument = await TileDocument.deterministic(ceramic, {
+        deterministic: true,
+        family: 'testCapabilities1',
+      })
+      const streamId = deterministicDocument.id
+      const didKeyWithCapability = await addCapToDid(wallet, didKey, `ceramic://${streamId.toString()}`)
+      ceramic.did = didKeyWithCapability
+
+      await deterministicDocument.update({ foo: 'bar' }, null, {
+        asDID: didKeyWithCapability,
```
Comment (ukstv) [inline packages/canary-integration/src/__tests__/capability.test.ts:335]: @zachferland Do I miss something or `asDID` is not necessary here?

**Review by ukstv — APPROVED** [review]
🏇

File: packages/canary-integration/src/__tests__/capability.test.ts (hunk 1 above)
Comment (zachferland) [reply packages/canary-integration/src/__tests__/capability.test.ts]: yup! removed

**Pull request merged**